### PR TITLE
core: add real transport test for retry buffer limit

### PR DESCRIPTION
**Pull request by dapengzhang0**
The unit tests in `RetriableStreamTest` do not really test buffer limit from end to end, because the buffer limit is implemented using `ClientStreamTracer.Factory`, and the tracer callback `outboundMessageSize()` is only triggered in `AbstractClientStream` after message serialization. In fact, it was broken without failing any existing tests (https://github.com/grpc/grpc-java/pull/8343#discussion_r677612379)

This PR adds a retry buffer limit test that runs through the `AbstractClientStream` code path.

### CODE REVIEW

File: core/src/test/java/io/grpc/internal/RetriableStreamTest.java [hunk 1]
```diff
@@ -972,6 +1006,95 @@ public void perRpcBufferLimitExceededDuringBackoff() {
...
+    // 2nd attempt received
+    serverCall = serverCalls.poll(5, TimeUnit.SECONDS);
+    serverCall.request(2);
+    verify(mockCallListener, never()).onClose(any(Status.class), any(Metadata.class));
+    // send one more message, should exceed buffer limit
+    call.sendMessage(message);
+    // let attempt fail
+    serverCall.close(
+        Status.UNAVAILABLE.withDescription("2nd attempt failed"),
+        pushBackMetadata);
```
Comment (ejona86) [inline core/src/test/java/io/grpc/internal/RetriableStreamTest.java:1091]: This should be a new Metadata instance.

**Review by ejona86 — APPROVED** [review]
Looks fine, but could we avoid putting it in core because of the netty dependency? Can we put it in netty or interop-testing instead? It seems `RetriableStream.GRPC_RETRY_PUSHBACK_MS` is the only part that benefits from being in core, and that can be replaced pretty easily.

File: core/src/test/java/io/grpc/internal/RetriableStreamTest.java (hunk 1 above)
Comment (dapengzhang0) [reply core/src/test/java/io/grpc/internal/RetriableStreamTest.java]: Done.

**Comment by dapengzhang0** [comment]
> Looks fine, but could we avoid putting it in core because of the netty dependency? Can we put it in netty or interop-testing instead?

Moved to interop-testing.

**Pull request merged**